### ISSUE TITLE
Gitflow 

### ISSUE THREAD

**Issue by longnd** [nvminhtue/coding_challenge#19]
First Kudos for using PRs and merging code bit by bit as it gives us good insight into your process. I notice some issues that I would like to hear your opinions on:

- Regarding branching, I noticed that you did not follow the standard git-flow: https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow. One of the main point is to have 2 main branches: `main` (master) and `develop`
The workflow you followed: branching out from the `master` branch and merging back to that branch is rather too simple in most cases. Is this the workflow you're applying in your current job? 

- Git commits: some PRs have giant commits, e.g:
  - this #5 PR has just a single big [single commit](https://github.com/nvminhtue/coding_challenge/pull/5/commits/c4fe048ecc47f5e4ef8e10105252c732ca38edca)
  - same on the #6 PR with [this commit](https://github.com/nvminhtue/coding_challenge/pull/6/commits/195ed5f139d7a902bfb4753f557d633115f929b5)
  - PR #8 - [commit](https://github.com/nvminhtue/coding_challenge/pull/8)
  
  ideally, the changes should be committed regularly to avoid losing any massive amount of changes, it also helps the code reviewers (not only in this code challenge) follow your thought easier.  

  Besides that, each commit should stage the changes for a group of files that correspond to the same change only, not massive changes, e.g. this commit f08ab0c `[BE] Add users table` not only defined the **users** table but also the **UserEntity**, **UserService**, changes for the Circle CI config

**Comment by nvminhtue** [comment]
@longnd
- For the git-flow, I am getting familiar with my current company's git-flow based on feature branches. I consider my working app as one feature and every PR should be merged on the feature branch instead of develop - main/master feature. My first thought was I will not include CD so there's no reason to create a develop/feature branch because it's should merge as one before deploying. I think right now it's kind of no longer properly, and it's should have a middle branch before merging to master once.

- In the second comment, I will not try to excuse these giant commits, that's my mistake but I want to clarify though. Initially, I broke tasks and each took around 4-8 hours of working/implementing, hence if you take a look at merging history it's around 1-3 merged PR a day. The very first PR included file creating, libraries lock files, and mark-up files and unit testings with chained as a flow which I already planned

I was overwhelmed with this challenge and my current-company workload which forces me to change the working context and let the commit becomes giant because I don't want to push a not completed function (my biggest mistake)

- The last word, I do think I did a messy merging history with massive changes on each commit, I ensure that I can do it better and more formally when adapting to the enormous workload like the past week, Once again, I apologize if these mistakes bother you.
